### PR TITLE
K8SPXC-462 add readiness probe for HAproxy

### PR DESCRIPTION
**Pull request by hors**
[K8SPXC-462](https://jira.percona.com/browse/K8SPXC-462)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| scaling | passed |
| recreate | passed |
| big-data | passed |
| storage | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| limits | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| security-context | passed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| scheduled-backup | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| demand-backup-encrypted-with-tls | failed |
| tls-issue-cert-manager-ref | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

File: pkg/pxc/app/statefulset/haproxy.go [hunk 1]
```diff
@@ -132,10 +142,20 @@ func (c *HAProxy) SidecarContainers(spec *api.PodSpec, secrets string, cr *api.P
...
 	container := corev1.Container{
 		Name:            "pxc-monit",
 		Image:           spec.Image,
 		ImagePullPolicy: spec.ImagePullPolicy,
+		ReadinessProbe: app.Probe(&corev1.Probe{
+			InitialDelaySeconds: redinessDelay,
+			TimeoutSeconds:      1,
+			PeriodSeconds:       5,
+			FailureThreshold:    5,
+		}, "/usr/local/bin/readiness-check.sh"),
```
Comment (delgod) [inline pkg/pxc/app/statefulset/haproxy.go:158]: why it is needed in the side car?

File: pkg/pxc/app/statefulset/haproxy.go [hunk 2]
```diff
@@ -53,10 +53,20 @@ func NewHAProxy(cr *api.PerconaXtraDBCluster) *HAProxy {
...
+	}
 	appc := corev1.Container{
 		Name:            haproxyName,
 		Image:           spec.Image,
 		ImagePullPolicy: spec.ImagePullPolicy,
+		ReadinessProbe: app.Probe(&corev1.Probe{
+			InitialDelaySeconds: redinessDelay,
+			TimeoutSeconds:      1,
+			PeriodSeconds:       5,
+			FailureThreshold:    5,
```
Comment (delgod) [inline pkg/pxc/app/statefulset/haproxy.go:68]:
```suggestion
			FailureThreshold:    3,
```
what do you think?

**Review by delgod — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| haproxy | passed |
| storage | passed |
| recreate | passed |
| big-data | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring | failed |
| upgrade-consistency | passed |
| demand-backup | passed |
| security-context | passed |
| monitoring-2-0 | failed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-cert-manager-ref | passed |

File: pkg/pxc/app/statefulset/haproxy.go (hunk 1 above)
Comment (hors) [reply pkg/pxc/app/statefulset/haproxy.go]: I have deleted it

File: pkg/pxc/app/statefulset/haproxy.go (hunk 2 above)
Comment (hors) [reply pkg/pxc/app/statefulset/haproxy.go]: I think it is ok.

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| big-data | passed |
| scaling | passed |
| haproxy | passed |
| recreate | passed |
| scaling-proxysql | passed |
| self-healing | failed |
| init-deploy | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| security-context | passed |
| demand-backup | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| demand-backup-encrypted-with-tls | passed |

**Pull request merged**